src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INSTALL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    public MyHashMap() {
+        this.capacity = INSTALL_CAPACITY;
+        this.threshold = (int) (INSTALL_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+
+        @Override
+        public String toString() {
+            return key + "=" + value;
+        }
+    }
+
+    private static int hash(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode());
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            table = resize();
+        }
+        putValue(new Node<>(hash(key), key, value), table);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> searchingNode = getNode(key);
+        return searchingNode == null ? null : searchingNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> getNode(K key) {
+        for (Node<K, V> headNode : table) {
+            Node<K, V> searchingNode = headNode;
+            if (searchingNode != null) {
+                while (searchingNode != null) {
+                    if (searchingNode.hash == hash(key)
+                            && (searchingNode.key == key
+                            || (searchingNode.key != null
+                            && searchingNode.key.equals(key)))) {
+                        return searchingNode;
+                    }
+                    searchingNode = searchingNode.next;
+                }
+            }
+        }
+        return null;
+    }
+
+    private Node<K, V>[] resize() {
+        size = 0;
+        capacity = capacity << 1;
+        threshold = threshold << 1;
```

Comment:
May be better to use static final increament  to change pacacity and threshold?
But it my point of view. 
If you want you can not do it.

Reply:
But how? If we increamet dy constant we will have 16 + "16" -> 32 + "16" -> 48... but expect 16->32->64....

Reply:
You can write:
threshold = threshold * MULTIPLIER;

and in the class field:
public static final int MULTIPLIER = 2;

But it up to you.